src/CodeCompliance/Domain/Checks/Filters/PluginFilter.php

```diff
@@ -46,13 +56,17 @@ public function filter(array $sources): array
      */
     protected function isPlugin(ReflectionClass $class): bool
     {
-        $pattern = '/.*Plugin$/';
-        $parent = $class->getParentClass();
-        $className = $class->getShortName();
-        if (!$parent) {
-            return false;
-        }
-
-        return (preg_match($pattern, $className)) && (preg_match($pattern, $parent->getShortName()));
+        return preg_match(static::PLUGIN_DIR_NAME, $this->reverseSlash($class->getName())) ||
+            preg_match(static::PLUGIN_CLASS_SUFFIX, $class->getShortName());
+    }
+
+    /**
+     * @param string $source
+     *
+     * @return string
+     */
+    protected function reverseSlash(string $source): string
+    {
+        return str_replace('\\', '/', $source);
```

Comment:
I think the more proper way is to use DIRECTORY_SEPARATOR instead of '/', because in different OS there is different directory separator. But as I understand the requirements of Spryker system is Linux only, so it's fine for me.